transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollServerChannel.java

```diff
@@ -75,11 +76,19 @@ public void connect(SocketAddress socketAddress, SocketAddress socketAddress2, C
         @Override
         void epollInReady() {
             assert eventLoop().inEventLoop();
+            boolean edgeTriggered = isFlagSet(Native.EPOLLET);
+
+            final ChannelConfig config = config();
+            if (!readPending && !config.isAutoRead() && !edgeTriggered) {
```

Comment:
Actually, my suggestion was dumb. It only proves we will end up with calling `config()` in either cases.